### DISCUSSION TITLE
Dynamic Recipients field - prepopulate the selected value from a GET parameter?

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [solspace/craft-freeform#957] [Questions & Answers] [answered]
Hiya,

Is it possible somehow to prepopulate the selected value of a Dynamic Recipients field from a GET parameter, but _without_ revealing the actual email value? It works ok if I use `?fieldHandle=test@example.com` but naturally that somewhat defeats the point of being able to keep the email addresses secret.

James

**Comment by kjmartens** [comment, marked as answer]
I don't think this is possible at this time, but it's certainly something that should be doable in Freeform 5 (in beta). The Dynamic Recipients field type has been replaced by a "User Select" feature that acts as a layer ontop of regular fields. So for example, if you want a "Dynamic Recipients" Dropdown field, in Freeform 5 you would instead create a regular Dropdown field, give it options like:

- `sales` / `Sales`
- `support` / `Support`
- etc

And then configure the "User Select" layer for it inside the form builder, where you are basically telling Freeform that the `sales` option should email `sales@site.com`, etc. So then, your GET param could just include `?fieldHandle=sales`. 🙂 

More info:
https://docs.solspace.com/craft/freeform/v5/forms/email-notifications/#user-select

**Reply by thisisjamessmith** [reply]
Thanks for the quick response! Given that we're using a mostly manual template-based output I think I can instead prepopulate it  using Twig and some manipulation of the `label` to make it url-safe.
